langchain/vectorstores/chroma.ts

```diff
@@ -0,0 +1,187 @@
+import fs from "fs/promises";
+import path from "path";
+import type {
+  ChromaClient as ChromaClientT,
+} from "chromadb";
+import { ChromaClient as ChromaClientO } from "chromadb";
+
+import { Embeddings } from "../embeddings/base";
+
+import { DocStore, SaveableVectorStore } from "./base";
+import { Document } from "../document";
+
+const COLLECTION_NAME = "langchain-collection-2";
+
+let ChromaClient: typeof ChromaClientT | null = null;
+
+try {
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  ({ ChromaClient } = require("chromadb"));
+} catch {
+  // ignore error
+}
+ChromaClient = ChromaClientO;
+
+export interface ChromaLibArgs {
+  space: string;
+  numDimensions?: number;
+}
+
+export class Chroma extends SaveableVectorStore {
+  index?: ChromaClientT;
+  
+  docstore: DocStore;
+
+  args: ChromaLibArgs;
+
+  constructor(
+    args: ChromaLibArgs,
+    embeddings: Embeddings,
+    docstore: DocStore,
+    index?: ChromaClientT
+  ) {
+    super(embeddings);
+    this.index = index;
+    this.args = args;
+    this.embeddings = embeddings;
+    this.docstore = docstore;
+  }
+
+  async addDocuments(documents: Document[]): Promise<void> {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    await this.addVectors(await this.embeddings.embedDocuments(texts), documents);
+  }
+
+  async addVectors(vectors: number[][], documents: Document[]) {
+    if (vectors.length === 0) {
+      return;
+    }
+    if (!this.index) {
+      if (this.args.numDimensions === undefined) {
+        this.args.numDimensions = vectors[0].length;
+      }
+      if (ChromaClient === null) {
+        throw new Error(
+          "Please install chromadb as a dependency with, e.g. `npm install -S chromadb`"
+        );
+      }
+      this.index = new ChromaClient("http://localhost:8000");
+      try {
+        await this.index.createCollection(COLLECTION_NAME);
```

Comment:
Wouldn't we want a different collection id for each instance?

Reply:
yeah - good point. there was no affordance for this obviously with the narrow `hnsw` version, but that should definitely be included here.

Reply:
This has been addressed. thanks for noting this!